### ISSUE TITLE
Issues with local registration when using nested_distributed_piecewise_alignment_pipeline

### ISSUE THREAD

**Issue by jboulanger91** [GFleishman/CircuitSeeker#2]
Hi Greg, 

I have been trying to optimize the parameters space to get single-cell resolution in an EM/2P pair in a fully automated way. I have attached an image that shows the output. 

![swirls_cs](https://user-images.githubusercontent.com/62715309/187236507-4d3d103e-0d3f-48a6-b6b3-a71f60f06e69.PNG)

Here you can see that some swirls appear (left red circle) although the shearing values have been kepts to 0. This tends to appear at high and low volume dicing. 
Also, conversely to the previous versions erronous deformations tend to appear when overlapping data is missing rather that keeping the original deformation (left red circle). The control over this parameter seem to be gone in the last posted version.  

I have used the following parameters: 
`block_schedule = [ [tuple(np.round(np.array(fix.shape) / 64).astype(int))], [tuple(np.round(np.array(fix.shape) / 32).astype(int))]]`
`randomiter=2500`

`    parameter_schedule= `
` [{'random_kwargs':{'max_translation':max_trans,
                          'max_rotation':2. * np.pi/180.,
                          'max_scale':1.02,
                          'max_shear':.00,
                          'random_iterations':randomiter,
                          'affine_align_best':10,
                          'iterations':24,},
         'affine_kwargs':{},
         'deform_kwargs':{'control_point_spacing':5.0,
                        'control_point_levels':[2, 4, 6],
                        'iterations':100,
                        'metric':'MI',},  
        }, 
`
`
        {'random_kwargs':{'max_translation':max_trans,
                          'max_rotation':2. * np.pi/180.,
                          'max_scale':1.02,
                          'max_shear':.00,
                          'random_iterations':randomiter,
                          'affine_align_best':10,
                          'iterations':24,},
         'affine_kwargs':{},
         'deform_kwargs':{'control_point_spacing':5.0,
                        'control_point_levels':[2, 4],
                        'iterations':100,
                        'metric':'MI',},  
        }
    ]`

I am sorry for this overwhelming message ... 
Happy to give you more clarifications to reproduce this issue. 

Thanks a lot, 
Jon

**Comment by jboulanger91** [comment]
The 'deform_kwargs' are causing the swirls but great alignement can be achieved w/out it.

**Comment by jboulanger91** [comment]
One thing that could help in the documentation is what these variables in  

`nested_distributed_piecewise_alignment_pipeline` are doing: 
`bins=64,
        shrink_factors=[1,],
        smooth_sigmas=[6.0,],
        iterations=400,
        learning_rate=0.1,
        max_step=0.1,
        estimate_learning_rate='never',
        callback=lambda irm: None,`

Thanks a lot again, 
J
